client/pages/result.jsx

```diff
@@ -89,6 +91,20 @@ export default class Result extends React.Component {
       });
   }
 
+  handleBookmark() {
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ state: this.state, userId: this.context.user.userId })
```

Comment:
Instead of putting a user id in the body (where anybody can fake their identity with a tool like HTTPie), you should send an `X-Access-Token` header and have your auth middleware extract the user info from the token in a secure manner.